File: algorithms/1_easy/M/00746_min_cost_climbing_stairs/solution.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        prev, prev_prev = (0, 0)

        for c in cost:
            curr = c + min(prev, prev_prev)
            prev_prev, prev = (prev, curr)

        return min(prev, prev_prev)

File: algorithms/1_easy/M/00746_min_cost_climbing_stairs/solution.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let (mut prev, mut prev_prev) = (0_i32, 0_i32);

        for c in cost {
            let curr = c + prev.min(prev_prev);
            (prev_prev, prev) = (prev, curr);
        }

        prev.min(prev_prev)
    }
}
